### PR TITLE
fix TestHttpProbe/returns_the_status_code_if_the_request_is_successful

### LINKED ISSUE #52: Flaky test: `TestHttpProbe/returns_the_status_code_if_the_request_is_successful`
Occurrences:
- https://github.com/jesusprubio/up/actions/runs/9686542637/job/26729185707
- https://github.com/jesusprubio/up/actions/runs/9686388900/job/26728671722

### PR DESCRIPTION
**Pull request by abdoroot**
Fixes #52.

### CODE REVIEW

**Comment by jesusprubio** [comment]
Thank you! I would prefer to merge a proper solution instead this workaround.

The problem here is the function `newTestHTTPServer` returning before the server is listening. Because the call to `ListenAndServe` is made inside a goroutine: https://github.com/jesusprubio/up/blob/520cd9e9fc0c35e2b1be6f70106bf6ed222bb54a/pkg/protocols_test.go#L66-L67

Please check this solution. I think it should work: https://stackoverflow.com/a/53332769

**Comment by abdoroot** [comment]
thanks for suggestion it work

File: pkg/protocols_test.go [hunk 1]
```diff
@@ -59,19 +59,30 @@ func TestHttpProbe(t *testing.T) {
 // Creates an HTTP server for testing.
 func newTestHTTPServer(t *testing.T) *http.Server {
 	hostPort := net.JoinHostPort("127.0.0.1", "8080")
-	server := &http.Server{Addr: hostPort}
-	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+	mux := http.NewServeMux()
+	server := &http.Server{
```
Comment (jesusprubio) [inline pkg/protocols_test.go:63]: Please, use one line to define `server`.

File: pkg/protocols_test.go [hunk 2]
```diff
@@ -59,19 +59,30 @@ func TestHttpProbe(t *testing.T) {
...
+		Addr: hostPort,
+		Handler:mux,
+	}
+
```
Comment (jesusprubio) [inline pkg/protocols_test.go:67]: Please, remove the empty lines to keep the formatting consistent.

File: pkg/protocols_test.go [hunk 3]
```diff
@@ -59,19 +59,30 @@ func TestHttpProbe(t *testing.T) {
...
+	l, err := net.Listen("tcp", hostPort)
+	if err != nil {
+		t.Fatalf("create listener %v", err)
+	}
+
 	go func() {
-		err := server.ListenAndServe()
-		if err != nil && err != http.ErrServerClosed {
+		if err := server.Serve(l); err != nil && err != http.ErrServerClosed {
```
Comment (jesusprubio) [inline pkg/protocols_test.go:78]: Use two different lines. Also to keep the formatting consistent with the codebase.

File: pkg/protocols_test.go [hunk 4]
```diff
@@ -59,19 +59,30 @@ func TestHttpProbe(t *testing.T) {
 // Creates an HTTP server for testing.
 func newTestHTTPServer(t *testing.T) *http.Server {
 	hostPort := net.JoinHostPort("127.0.0.1", "8080")
-	server := &http.Server{Addr: hostPort}
-	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+	mux := http.NewServeMux()
```
Comment (jesusprubio) [inline pkg/protocols_test.go:62]: We don't need a mux for this simple scenario. The line 63 should be enough.

**Review by jesusprubio — CHANGES_REQUESTED** [review]
Thank you. Only a couple of formatting changes are we are ready to merge. :)

File: pkg/protocols_test.go (hunk 4 above)
Comment (abdoroot) [reply pkg/protocols_test.go]: okay got it

Comment (jesusprubio) [reply pkg/protocols_test.go]: Thank you :)

**Pull request merged**